widget/textgrid.go

```diff
@@ -310,6 +309,135 @@ func (t *TextGrid) refreshCell(row, col int) {
 	r.refreshCell(row, col)
 }
 
+// ForRange iterates over a range of cells and rows within a TermGrid, optionally applying a function to each cell and row.
+//
+// Parameters:
+// - blockMode (bool): If true, the iteration is done in block mode, meaning it iterates through rows and applies the cell function for each cell in the specified column range.
+// - startRow (int): The starting row index for the iteration. Rows are 0-indexed.
+// - startCol (int): The starting column index for the iteration within the starting row. Columns are 0-indexed.
+// - endRow (int): The ending row index for the iteration.
+// - endCol (int): The ending column index for the iteration within the ending row.
+// - eachCell (func(cell *widget.TextGridCell)): A function that takes a pointer to a TextGridCell and is applied to each cell in the specified range. Pass `nil` if you don't want to apply a cell function.
+// - eachRow (func(row *widget.TextGridRow)): A function that takes a pointer to a TextGridRow and is applied to each row in the specified range. Pass `nil` if you don't want to apply a row function.
```

Comment:
This is misleading - it implies that it will be called for the first row, but it isn't